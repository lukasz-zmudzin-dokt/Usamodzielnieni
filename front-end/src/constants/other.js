const DEFAULT_INPUT = "-- Wybierz --";

export { DEFAULT_INPUT };
