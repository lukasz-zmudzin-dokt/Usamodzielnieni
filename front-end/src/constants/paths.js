const paths = {
    DASHBOARD: "/",
    CV_CREATOR: "/cvEditor",
    CV_EDITOR: "/cvEditor/:id",
    REGISTER: "/newAccount/",
    REGISTER_ADMIN: "/newAccount/:role(staff)",
    FOOTER: "/footer",
    LOGIN: "/login",
    USER: "/user",
    OFFER_FORM: "/offerForm/:id?",
    JOB_OFFERS: "/jobOffers",
    CV_APPROVAL: "/cvApproval",
    MY_OFFERS: "/myOffers",
    CONTACT_PAGE: "/contact",
    BLOG_POST: "/blog/blogpost/:id",
    BLOG_PAGE: "/blog",
    BLOG_FORM: "/blog/newPost",
    BLOG_EDIT: "/blog/newPost/:id",
    JOB_OFFER_DETAILS: "/jobOffers/:id",
    MY_CVS: "/myCVs",
    USER_APPROVAL: "/userApproval",
    OFFER_APPROVAL: "/offerApproval"
};
export {paths};