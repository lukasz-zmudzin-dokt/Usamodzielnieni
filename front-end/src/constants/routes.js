import {
  UserProfilePage,
  LoginPage,
  RegisterPage,
  Footer,
  CVEditorPage,
  Menu,
  OfferForm,
  MyOffersPage,
  CVApprovalPage
} from "Views";

const paths = {
  DASHBOARD: "/",
  CV_EDITOR: "/cvEditor",
  REGISTER: "/newAccount",
  FOOTER: "/footer",
  LOGIN: "/login",
  USER: "/user",
  OFFER_FORM: "/offerForm",
  JOB_OFFERS: "/jobOffers",
  CV_APPROVAL: "/cvApproval",
  MY_OFFERS: "/myOffers"
};

export const userTypes = {
  STANDARD: "Standard",
  STAFF: "Staff",
  EMPLOYER: "Employer"
};

export default [
  {
    path: paths.DASHBOARD,
    component: Menu,
    exact: true
  },
  {
    path: paths.CV_EDITOR,
    component: CVEditorPage,
    isPrivate: true,
    type: userTypes.STANDARD
  },
  {
    path: paths.REGISTER,
    component: RegisterPage,
    exact: true
  },
  {
    path: paths.FOOTER,
    component: Footer,
    exact: true
  },
  {
    path: paths.LOGIN,
    component: LoginPage,
    exact: true
  },
  {
    path: paths.USER,
    component: UserProfilePage,
    isPrivate: true,
    type: undefined // jeżeli jest undefined to znaczy że jest dostępne dla wszystkich typów konta
  },
  {
    path: paths.JOB_OFFERS,
    component: UserProfilePage, // tu trzeba zmienić komponent
    isPrivate: true,
    type: userTypes.EMPLOYER
  },
  {
    path: paths.OFFER_FORM,
    component: OfferForm,
    isPrivate: true,
    type: userTypes.EMPLOYER
  },
  {
    path: paths.CV_APPROVAL,
    component: CVApprovalPage,
    isPrivate: true,
    type: userTypes.STAFF
  },
  {
    path: paths.MY_OFFERS,
    component: MyOffersPage,
    isPrivate: true,
    type: userTypes.EMPLOYER
  }
];

export { paths };
