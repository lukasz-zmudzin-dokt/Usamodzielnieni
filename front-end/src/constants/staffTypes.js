export const staffTypes = {
  VERIFICATION: "staff_verification",
  CV: "staff_cv",
  JOBS: "staff_jobs",
  BLOG_CREATOR: "staff_blog_creator",
  BLOG_MODERATOR: "staff_blog_moderator",
};
