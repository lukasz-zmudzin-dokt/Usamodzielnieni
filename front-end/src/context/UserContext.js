import React, { useState } from "react";
import Cookies from "universal-cookie";

export const UserContext = React.createContext({
  type: undefined, // Employer,Standard,Staff
  token: undefined,
  login: () => {},
  logout: () => {}
});
const cookies = new Cookies();

export const UserProvider = props => {
  const [type, setType] = useState(cookies.get("type"));
  const [token, setToken] = useState(cookies.get("token"));
  const user = {
    token,
    type,
    login: (newToken, newType) => {
      cookies.set("token", newToken, { path: "/" });
      cookies.set("type", newType, { path: "/" });
      setToken(newToken);
      setType(newType);
    },
    logout: () => {
      cookies.remove("token", { path: "/" });
      cookies.remove("type", { path: "/" });
      setToken(undefined);
      setType(undefined);
    }
  };
  return <UserContext.Provider value={user} {...props} />;
};
