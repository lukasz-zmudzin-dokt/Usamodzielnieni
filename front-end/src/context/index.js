export { UserContext, UserProvider } from "context/UserContext";
export { AlertContext, AlertProvider } from "context/AlertContext";
export { ChatContext, ChatProvider } from "context/ChatContext";
export {
  NotificationsContext,
  NotificationsProvider,
} from "context/NotificationsContext";
