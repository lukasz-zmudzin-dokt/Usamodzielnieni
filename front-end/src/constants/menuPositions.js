import { userTypes } from "./userTypes";
import { staffTypes } from "./staffTypes";
import { paths } from "./paths";

export default [       
    {
        name: "Kreator CV",
        path: paths.CV_CREATOR,
        allowed: [userTypes.STANDARD],
        verified: true
    },
    {
        name: "Moje CV",
        path: paths.MY_CVS,
        allowed: [userTypes.STANDARD],
        verified: true
    },
    {
        name: "Akceptacja CV",
        path: paths.CV_APPROVAL,
        allowed: [staffTypes.CV],
        verified: true
    },
    {
        name: "Oferty pracy",
        path: paths.JOB_OFFERS,
        allowed: [userTypes.STANDARD, userTypes.STAFF, userTypes.EMPLOYER]
    },
    {
        name: "Dodaj ofertę",
        path: paths.OFFER_FORM,
        allowed: [userTypes.EMPLOYER],
        verified: true
    },
    {
        name: "Moje oferty",
        path: paths.MY_OFFERS,
        allowed: [userTypes.EMPLOYER],
        verified: true
    },
    {
        name: "Blogi",
        path: paths.BLOG_PAGE,
        allowed: [userTypes.STANDARD, userTypes.STAFF, userTypes.EMPLOYER]
    },
    {
        name: "Dodaj post",
        path: paths.BLOG_FORM,
        allowed: [staffTypes.BLOG_CREATOR],
        verified: true
    },
    {
        name: "Telefony",
        path: paths.CONTACT_PAGE,
        allowed: undefined, //wszyscy są upoważnieni - zalogowani i niezalogowani
    },
    {
        name: "Kroki",
        path: paths.STEPS,
        allowed: undefined //wszyscy są upoważnieni - zalogowani i niezalogowani
    },
    {
        name: "Chat",
        path: paths.CHATS,
        allowed: [userTypes.STANDARD, userTypes.STAFF, userTypes.EMPLOYER],
        verified: true,
    }
];
