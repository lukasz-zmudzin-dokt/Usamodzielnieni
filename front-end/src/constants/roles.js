import { staffTypes } from "./staffTypes";

export const adminGroup = [
  {
    name: staffTypes.VERIFICATION,
    placeholder: "Weryfikacja użytkowników",
  },
  {
    name: staffTypes.CV,
    placeholder: "Weryfikacja CV",
  },
  {
    name: staffTypes.JOBS,
    placeholder: "Weryfikacja ofert pracy",
  },
  {
    name: staffTypes.BLOG_CREATOR,
    placeholder: "Kreator postów na blogu",
  },
  {
    name: staffTypes.BLOG_MODERATOR,
    placeholder: "Moderator bloga / designer",
  },
  {
    name: staffTypes.GUEST,
    placeholder: "Konto gościa",
  },
  {
    name: staffTypes.CHAT,
    placeholder: "Dostęp do czatu",
  },
];

export const commonGroup = [
  {
    name: "Podopiecznym",
  },
  {
    name: "Pracodawcą",
  },
];
