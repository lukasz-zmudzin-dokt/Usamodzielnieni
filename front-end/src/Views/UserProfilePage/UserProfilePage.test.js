import React from "react";
import {fireEvent, render, waitForElement} from "@testing-library/react";
import UserProfile from "Views/UserProfilePage/index.js";
import {MemoryRouter, Router} from "react-router-dom";
import {createMemoryHistory} from 'history';
import {UserContext} from "context/UserContext";
import {staffTypes} from "constants/staffTypes";

const renderWithRouter = (
    type,
    ui,
    {
      route = "/user",
      history = createMemoryHistory({ initialEntries: [route] }),
    } = {}
) => {
  let context = { type: 'Staff', data: {group_type: type}, token: 123 };
  return {
    ...render(
        <UserContext.Provider value={context}>
          <Router history={history}>{ui}</Router>
        </UserContext.Provider>
    ),
    history,
  };
};

describe("UserProfile", () => {
  let user = {
    id: 123,
    first_name: "Jan",
    last_name: "Kowalski",
    username: "jankowalski",
    email: "hello@world.com"
  };
  let apiFail;

  beforeAll(() => {
    global.fetch = jest.fn().mockImplementation((input, init) => {
      return new Promise((resolve, reject) => {
        if (apiFail) {
          resolve({status: 500});
        } else {
          switch(init.method) {
            case "GET":
              resolve({status: 200, json: Promise.resolve(user)});
              break;
            default:
              reject({});
              break;
          }
        }
      });
    });
  });

  beforeEach(() => {
    apiFail = false;
    jest.clearAllMocks();
  });

  it("should render correctly", () => {
    const {container} = render(
        <MemoryRouter>
          <UserProfile/>
        </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render alert on api fail', async () => {
    apiFail = true;
    const {getByText} = render(
        <MemoryRouter>
          <UserProfile/>
        </MemoryRouter>
    );

    await waitForElement(() => getByText("Wystąpił błąd", {exact: false}));

    expect(getByText("Wystąpił błąd", {exact: false})).toBeInTheDocument();
  });

  it('should render register button for staff reg', () => {
    const {history, getByText} = renderWithRouter( staffTypes.VERIFICATION,
        <UserProfile/>
    );

    expect(getByText('Zarejestruj administratora')).toBeInTheDocument();

    fireEvent.click(getByText('Zarejestruj administratora'));

    expect(history.location.pathname).toBe("/newAccount/staff", {exact: false})
  });

  it('should render cv button for staff cv', () => {
    const {history, getByText} = renderWithRouter( staffTypes.CV,
        <UserProfile/>
    );

    expect(getByText('Zobacz CV do akceptacji')).toBeInTheDocument();

    fireEvent.click(getByText('Zobacz CV do akceptacji'));

    expect(history.location.pathname).toBe("/cvApproval", {exact: false})
  });

  it('should render user list button', () => {
    const {history, getByText} = renderWithRouter( staffTypes.CV,
        <UserProfile/>
    );

    expect(getByText('Zobacz listę wszystkich użytkowników')).toBeInTheDocument();

    fireEvent.click(getByText('Zobacz listę wszystkich użytkowników'));

    expect(history.location.pathname).toBe("/userList", {exact: false})
  });

  it('should render no buttons', () => {
    const {queryByText} = render(
        <UserContext.Provider value={{type: "Standard"}}>
          <MemoryRouter>
            <UserProfile />
          </MemoryRouter>
        </UserContext.Provider>
    );

    expect(queryByText('Zobacz listę wszystkich użytkowników')).not.toBeInTheDocument();
    expect(queryByText('Zarejestruj administratora')).not.toBeInTheDocument();
    expect(queryByText('Zobacz CV do akceptacji')).not.toBeInTheDocument();
  })

});