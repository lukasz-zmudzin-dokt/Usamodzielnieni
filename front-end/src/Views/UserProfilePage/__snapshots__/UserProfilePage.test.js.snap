// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile should render correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="user_profile_card card"
    >
      <div
        class="user_card_title card-header"
      >
        <h3>
          Mój profil
        </h3>
      </div>
      <div
        class="card-body"
      >
        <div>
          UserBasicInfo
        </div>
      </div>
      <div
        class="list-group-flush list-group"
      >
        <div
          class="property list-group-item"
        >
          <div
            class="row"
          >
            <div
              class="col-sm-auto col-12"
            >
              <div
                class="property__title"
              >
                Nazwa użytkownika
                :
              </div>
            </div>
            <div
              class="col"
            />
          </div>
        </div>
        <div
          class="property list-group-item"
        >
          <div
            class="row"
          >
            <div
              class="col-sm-auto col-12"
            >
              <div
                class="property__title"
              >
                E-mail
                :
              </div>
            </div>
            <div
              class="col"
            />
          </div>
        </div>
        <div
          class="property list-group-item"
        >
          <div
            class="row"
          >
            <div
              class="col-sm-auto col-12"
            >
              <div
                class="property__title"
              >
                Rola na czacie
                :
              </div>
            </div>
            <div
              class="col"
            />
          </div>
        </div>
      </div>
      <div
        class="text-center card-body"
      >
        <div
          class="buttonsContainer"
        >
          <button
            class="btn btn-danger"
            type="button"
          >
            Usuń konto
          </button>
          <button
            class="btn btn-info"
            type="button"
          >
            Zmień hasło
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
